### ISSUE TITLE
Bad word filters with 'escaped' parenthesis cause errors as 1.8.19

### ISSUE THREAD

**Issue by bartico6** [mybb/mybb#3461]
Filter: "\()"

Error: Warning [2] preg_replace(): Compilation failed: unmatched parentheses at offset 32 - Line: 659 - File: inc/class_parser.php

Filter: "(\)"

Error: Warning [2] preg_replace(): Compilation failed: missing ) at offset 7 - Line: 659

**Comment by bartico6** [comment]
Appending: Before 1.8.18 filters with 'escaped' parenthesis worked fine.
The case for using those is certain variants of the lenny face which we filter out.
They worked fine until the update, which caused the engine to spam regex errors all over the site.

I also fixed the wrong version "blame", it was 1.8.18 and not 1.8.19, since 1.8.19's diff suggests the inc/class_parser.php file was not touched at all.

**Comment by Eldenroot** [comment]
@effone 

Yes, there were some changes for bad words filter feature, good catch
